modules/dxp/apps/portal-search-tuning/portal-search-tuning-blueprints-admin-web/src/main/resources/META-INF/resources/js/edit_blueprint/index.js

```diff
@@ -601,41 +601,46 @@ function EditBlueprintForm({
 				tab={tab}
 				tabs={TABS}
 			>
-				<ClayToolbar.Item>
-					<ClayButton
-						borderless
-						className={getCN({
-							active: showPreview,
-						})}
-						displayType="secondary"
-						onClick={() => {
-							setShowSidebar(false);
-							setShowPreview(!showPreview);
-						}}
-						small
-					>
-						{Liferay.Language.get('preview')}
-
-						{previewInfo.results.errors &&
-							!!previewInfo.results.errors.length && (
-								<span className="inline-item inline-item-after">
-									<ClayBadge
-										displayType="danger"
-										label={
-											previewInfo.results.errors.length
-										}
-									/>
-								</span>
-							)}
-					</ClayButton>
-				</ClayToolbar.Item>
+				{useMemo(
+					() => (
+						<ClayToolbar.Item>
+							<ClayButton
+								borderless
+								className={getCN({
+									active: showPreview,
+								})}
+								displayType="secondary"
+								onClick={() => {
+									setShowSidebar(false);
+									setShowPreview(!showPreview);
+								}}
+								small
+							>
+								{Liferay.Language.get('preview')}
+
+								{!!previewInfo.results.errors?.length && (
+									<span className="inline-item inline-item-after">
+										<ClayBadge
+											displayType="danger"
+											label={
+												previewInfo.results.errors
+													.length
+											}
+										/>
+									</span>
+								)}
+							</ClayButton>
+						</ClayToolbar.Item>
+					),
+					[showPreview, previewInfo.results.errors?.length]
+				)}
```

Comment:
Do you think this useMemo is worth any performance gain? Is this just preventing 1 button from re-rendering? It doesn't seem worth maintaining and more likely to have bugs by keeping track of `[showPreview, previewInfo.results.errors?.length]`.

Reply:
Ah ok! I'll take it out then.